src/Checker.ts

```diff
@@ -1,14 +1,30 @@
 import {ReasonGuard} from './ReasonGuard';
 import {NegatableGuard, buildNegatable} from './NegatableGuard';
+import {thenGuard} from './Combinators';
+import {isString} from './primitiveGuards';
 
 export type Checker<FROM> = (input: FROM) => string;
 
 export const checkerToGuard: <FROM, TO extends FROM, N extends FROM = FROM>(
 	checker: Checker<FROM>
-) => NegatableGuard<FROM, TO, N> = (checker) => buildNegatable(
-	() => getRawGuard(checker),
-	() => getRawNegation(checker)
-);
+	) => NegatableGuard<FROM, TO, N> = (checker) => buildNegatable(
+		() => getRawGuard(checker),
+		() => getRawNegation(checker)
+	);
+
+const uuidChecker = (input: string) => {
+	if (!!input
+			&& /[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}/.test(input)
```

Comment:
You should store this regex external to the checker so that it is cached, and isn't getting recompiled for each use.